src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java

```diff
@@ -266,4 +270,114 @@ public StringValueResponse terminateEvent(@PathVariable("procID") final Integer
     procurementEventService.terminateEvent(procId, eventId, type.getTerminationType(), principal);
     return new StringValueResponse("OK");
   }
+
+  @GetMapping("/{eventID}/responses/export")
+  public  ResponseEntity<StreamingResponseBody> exportAllResponses(
+          @PathVariable("procID") final Integer procId, @PathVariable("eventID") final String eventId,
+          HttpServletResponse response, final JwtAuthenticationToken authentication){
+
+    var principal = getPrincipalFromJwt(authentication);
+    log.info("getDocumentSummaries invoked on behalf of principal: {}", principal);
+
+    List<SupplierAttachmentResponse> supplierAttachmentResponseList=procurementEventService.getSupplierAttachmentResponses(procId, eventId);
+
+    StreamingResponseBody streamResponseBody =
+        out -> {
+          final ZipOutputStream zipOutputStream = new ZipOutputStream(response.getOutputStream());
+          ZipEntry zipEntry = null;
+
+          for (SupplierAttachmentResponse supplierAttachmentResponse :
+              supplierAttachmentResponseList) {
+
+            List<ParameterInfo> parameterInfoList =
+                supplierAttachmentResponse.getParameterInfoList();
+           
+            for (ParameterInfo parameterInfo : parameterInfoList) {
+              for (AttachmentInfo attachmentInfo : parameterInfo.getAttachmentInfoList()) {
+                var docAttachment =
+                    procurementEventService.downloadAttachment(
+                        Integer.parseInt(attachmentInfo.getAttachmentId()),
+                        attachmentInfo.getAttachmentName());
+
+                var filename=String.join("_",supplierAttachmentResponse.getSupplierId().toString(),attachmentInfo.getParameterId().toString(),supplierAttachmentResponse.getSupplierName());
+                filename=filename.concat(attachmentInfo.getAttachmentName().substring(attachmentInfo.getAttachmentName().indexOf('.')));
+
+                zipEntry =
+                    new ZipEntry(filename);
+
+                zipOutputStream.putNextEntry(zipEntry);
+                try (InputStream is = new ByteArrayInputStream(docAttachment.getData())) {
+                  IOUtils.copy(is, zipOutputStream);
+                }
+              }
+            }
+          }
+          // set zip size in response
+          response.setContentLength((int) (zipEntry != null ? zipEntry.getSize() : 0));
+
+          if (zipOutputStream != null) {
+            zipOutputStream.close();
+          }
+        };
+
+    response.setContentType("application/zip");
+    response.setHeader("Content-Disposition",
```

Comment:
Can we make use of the header constants provided by Spring, e.g. `HttpHeaders.CONTENT_DISPOSITION` ?

Reply:
using HttpHeaders  now where possible